libutils/hash_map.c

```diff
@@ -172,6 +172,40 @@ bool HashMapRemove(HashMap *map, const void *key)
     return false;
 }
 
+bool HashMapRemoveSoft(HashMap *map, const void *key)
+{
+    assert(map != NULL);
+
+    unsigned bucket = HashMapGetBucket(map, key);
+
+    /*
+     * prev points to a previous "next" pointer to rewrite it in case value need
+     * to be deleted
+     */
+
+    for (BucketListItem **prev = &map->buckets[bucket];
+         *prev != NULL;
+         prev = &((*prev)->next))
+    {
+        BucketListItem *cur = *prev;
+        if (map->equal_fn(cur->value.key, key))
+        {
+            map->destroy_key_fn(cur->value.key);
+            *prev = cur->next;
+            free(cur);
+            map->load--;
+            if ((map->load < map->min_threshold) && (map->size > map->init_size))
+
```

Comment:
white space

Reply:
? I don't see any white space problem in the above. Moreover, it's just a code copied from the `MapRemove()` function.

Reply:
I just though an additional newline between the condition and the if-block looked kind of weird. But I guess it's alright. 👍

Reply:
Yeah, like I said copy-pasted code. But I fixed it here at least. Good catch!